dNet/TcpTransportLayer.h

```diff
@@ -0,0 +1,158 @@
+#pragma once
+
+#include <iostream>
+#include <string>
+#include <boost/asio.hpp>
+#include <boost/asio/ssl.hpp>
+#include <thread>
+#include <mutex>
+#include <queue>
+#include <memory>
+#include <csignal>
+#include <unordered_map>
+#include <functional>
+#include <boost/functional/hash.hpp>
+
+#include "RakPeerInterface.h"
+#include "ReplicaManager.h"
+#include "NetworkIDManager.h"
+
+#include "TransportLayer.h"
+#include "TransportPeerInterface.h"
+#include "TcpSession.h"
+
+namespace std {
+	// SystemAddress hash specialization
+	template <>
+	struct hash<SystemAddress> {
+		std::size_t operator()(const SystemAddress& sysAddr) const {
+			const std::size_t hash1 = sysAddr.binaryAddress;
+			const std::size_t hash2 = sysAddr.port;
+			return (hash1 | (hash2 << 32));
+		}
```

Comment:
on platforms where size_t is 32 bits, this will result in just being hash1 always.  a 32 bit shift on a 4 byte value also is not a valid operation which is worth considering for a different hash value